fix: MLS group not visible and duplicated system messages - WPB-15681

jullianm opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-15681" title="WPB-15681" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-15681</a>  [General] MLS group created on Web (edge) does not show up on own iOS and invited member's Web clients
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove the jira markers to link tickets automatically -->


### Issue

**Issue 1**: When creating a MLS group with Web prod / edge and iOS app is in background, entering the foreground again will not show the MLS group. Reason is that we try to process the `conversation-create` event but we early exit since the conversation already exists locally.

It happens to work when app is in the foreground because usually `conversation-create` arrives before `conversation.mls-welcome`.

--> **Solution 1**: Removing the early exit when a conversation already exists so we can actually proceed the event.

**Issue 2**: The early exit was introduced in the first place to fix another issue related to duplicated system message when a participant is added to a conversation. See 'Causes' section in https://github.com/wireapp/wire-ios/pull/976 which provides some details regarding the reason of these duplicated messages. 

--> **Solution 2**: Don't remove participants when a `create-conversation` event is received, so when the app returns to the foreground and events are processed again, `member-join` event will be processed but it will not display a new system message since the participant was already added.

Note: Maybe it's not needed at all to call the `updateMembers` method for a `create-conversation` event since this is handled by the `member-join` or `member-leave` events however I'm not 100% confident this will not raise other bugs.

### Testing

For issue 1 : 
- Put iOS app in background
- Create MLS group conversation on Web prod / edge
- Put the iOS app in foreground, MLS group should appear

For issue 2: 
- on iOS app, create a MLS group and add a participant
- go to background
- go to foreground, there should be a single message `You added ..." 

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

netbe committed:
revert: simulcast support - WPB-15451 (#2390)

github-actions[bot] committed:
fix: enable Countly for AppStore builds - WPB-15419 (#2409)

Co-authored-by: Christoph Aldrian <christoph.aldrian@wire.com>

caldrian committed:
chore: move use case into WireDomainPkg - WPB-10347 (#2413)

github-actions[bot] committed:
fix: Connection request team bug - WPB-15275 (#2414)

Co-authored-by: Sam Wyndham <samwyndham@users.noreply.github.com>

jullianm committed:
fix: MLS add message decryption error - WPB-15646 (#2416)

netbe committed:
fix: crash persistent store first - WPB-15476 (#2408)

github-actions[bot] committed:
fix: Crowdin configuration - WPB-15715 (#2431)

netbe committed:
chore: bump CoreCrypto to 3.0.1 - WPB-15726 (#2437)

samwyndham committed:
fix: ServerTrustValidatorTests - WPB-15760 (#2440)

netbe committed:
fix: crash create app router - WPB-15683 (#2436)

caldrian committed:
fix: skip notification when away or busy - WPB-8946 (#2438)

netbe committed:
fix: mls system message - WPB-15129 (#2411)

netbe committed:
fix: rename system message - WPB-14824 (#2445)

netbe committed:
fix: crash dismissPresentedViewController - WPB-15345 (#2442)

jullianm committed:
fix MLS group not showing up, duplicated system message

jullianm committed:
remove useless code

jullianm committed:
lint and format

jullianm committed:
Merge remote-tracking branch 'origin/release/cycle-3.117' into fix/mls-group-not-visible-and-duplicated-messages

johnxnguyen reviewed (commented):
It looks like your PR has unrelated changes.

jullianm committed:
Revert "Revert "feat: simulcast support - WPB-11481 (#2251)""

This reverts commit 7310934f64250c543280566dcccf0bcdc53fa906.

jullianm commented:
because I branched out first from 3.116 and changed the target branch to 3.117 afterwards.
the unrelated commit was from this PR : https://github.com/wireapp/wire-ios/pull/2390
I reverted it. thanks @johnxnguyen

jullianm committed:
remove no longer needed UTs

jullianm commented on wire-ios-request-strategy/Tests/Sources/Payloads/Processing/ConversationEventPayloadProcessorTests.swift:
```diff
@@ -60,56 +60,6 @@ final class ConversationEventPayloadProcessorTests: MessagingTestBase {
 
     // MARK: - Process NewConversation Event
 
-    func testProcessPayload_NewConversation_IgnoredWhenConversationAlreadyExists() async throws {
```
no longer needed because we removed the early exit code

jullianm commented on wire-ios-request-strategy/Tests/Sources/Payloads/Processing/ConversationEventPayloadProcessorTests.swift:
```diff
@@ -60,56 +60,6 @@ final class ConversationEventPayloadProcessorTests: MessagingTestBase {
-        }
-    }
-
-    func testProcessPayload_NewConversation_IgnoredWhenConversationIDIsMissing() async throws {
```
no longer needed because we removed the related early exit code

netbe reviewed (approved):
LGTM;) I guess we don't have to report these changes to new code, @jullianm  ?

jullianm committed:
apply fixes to new WireDomain code

jullianm commented on wire-ios-request-strategy/Sources/Payloads/Processing/ConversationEventPayloadProcessor.swift:
```diff
@@ -528,7 +517,7 @@ struct ConversationEventPayloadProcessor {
             updateAttributes(from: payload, for: conversation, context: context)
             updateMetadata(from: payload, for: conversation, context: context)
-            updateMembers(from: payload, for: conversation, context: context)
+            updateMembers(from: payload, for: conversation, shouldRemoveParticipants: false, context: context)
```
yes, we no longer remove participants for conversation creation events (to fix the issue mentioned in PR description) but there's still the code that adds participants (in `updateMembers` method)
member join event adds participants and member leave removes them.
I'm not 100% sure it is needed at all to call `updateMembers` method here but I'm afraid if we do it'll raise other bugs.

jullianm committed:
Merge branch 'release/cycle-3.117' into fix/mls-group-not-visible-and-duplicated-messages
